opnode/rollup/driver/fake_chain.go

```diff
@@ -117,6 +117,16 @@ func (m *fakeChainSource) L1BlockRefByNumber(ctx context.Context, l1Num uint64)
 	return m.l1s[m.l1reorg][l1Num], nil
 }
 
+func (m *fakeChainSource) L1BlockRefByHash(ctx context.Context, l1Hash common.Hash) (eth.L1BlockRef, error) {
+	m.log.Trace("L1BlockRefByHash", "l1Hash", l1Hash, "l1Head", m.l1head, "reorg", m.l1reorg)
```

Comment:
Does this need a mutex?

Reply:
I don't believe so. There is a data race in the driver test (and only test), but I think it's related to reading state variables.